documentation/user_docs/docs/manual/Simulation.rst

```diff
@@ -2,305 +2,398 @@
+  dataset may contain gaps due to the trajectory of the detectors through
+  reciprocal space, but you may wish to simulate the scattering even in the
+  gaps. This option applies in the case of dnd objects, but not sqw objects.
 
-dispersion
-==========
+sqw_eval
+========
 
-Calculate dispersion relation for dataset or array of datasets.
+.. code-block:: matlab
 
-::
+   wout = sqw_eval(win, @my_sqw_func, p);
+   wout = sqw_eval(win, @my_sqw_func, p, 'all');
 
-   wdisp = dispersion (win, dispreln, p)            % dispersion only
+The syntax for ``sqw_eval`` is almost identical to that of ``func_eval``. The
+only difference is the form of the function required ``my_sqw_func``,
 
-   [wdisp,weight] = dispersion (win, dispreln, p)   % dispersion and spectral weight
+.. warning::
 
+   The function supplied to ``sqw_eval`` must be of the form:
 
-The output dataset (or array of data sets), ``wdisp``, will retain only the Q axes, and the signal array(s) will contain the values of energy along the Q axes. If the dispersion relation returns the spectral weight, this will be placed in the error array (actually the square of the spectral weight is put in the error array). In the case when the dispersion has been calculated on a plane in momentum (i.e. wdisp is IX_datset_2d) then the plot function
+   .. code-block:: matlab
 
-``ds2`` (for draw surface from two arrays)
+      weight = my_sqw_func(qh, qk, ql, en, p)
 
-::
+   where ``qh, qk, ql, en`` are arrays that contain the co-ordinates in
+   :math:`h, k, l` and energy of each point in the dataset, irrespective of the
+   dimensionality of the representative ``dnd``.
 
-   ds2(wdisp)
+- ``p`` is a row-vector containing the parameters required by the
+  function.
 
+.. note::
 
-will plot a surface with the z axis as energy and coloured according to the spectral weight.
+   These could be the values of exchange constants, intensity scale factor, or
+   temperature, for example.
 
-**If you wish to overplot a dispersion relation on top of, for example, a Q-E slice from your data, then you would use:**
+One would generally use ``sqw_eval`` in preference to ``func_eval`` if, for
+example, one had a model of the spin-wave cross-section for magnetic scattering.
 
-::
+dispersion
+==========
 
-   plot(my_qe_slice)
+Calculate dispersion relation for dataset or array of datasets.
 
-   ploc(wdisp)       % for plot line on current
+.. code-block:: matlab
 
+   [wdisp,weight] = dispersion(win, dispreln, p)   % dispersion and spectral weight
 
-Note that in the above there must not be a *keep* command between plotting the Q-E slice and plotting the dispersion, since the ``ploc`` command works on the *current* figure.
+The output dataset (or array of data sets), ``wdisp``, will retain only the Q
+axes, and the signal array(s) will contain the values of energy along the Q
+axes. If the dispersion relation returns the spectral weight, this will be
```

Comment:
```suggestion
points. If the dispersion relation returns the spectral weight, this will be
```
A bit of a stab in the dark, happy to be corrected on this.